### DISCUSSION TITLE
Airflow image 2.5.3:python3.8 has CVSS 9.8 Vuln that has a fix version

### DISCUSSION THREAD

**Discussion by ahipp13** [apache/airflow#30604] [Q&A] [answered]
Hello,

I am writing this because I am very stuck and in need of some help. I am trying to get rid of CVSS 9.8 vulns that have a fix version from our Airflow Image. I downloaded a base airflow 2.5.3 python3.8 airflow image, and it looks like there is one CVE with a fix version:

CVE-2021-38297

It says there are currently 2 libraries on the image that are causing this:

github.com/golang/go:1.16.8
github.com/golang/go:1.15.9

I have opened a cmd on our kubernetes pods to see and if I can find these on the pods, and I cannot find either of them... They need to be updated to 1.16.9 or 1.17.2. So my question is, what is installing go on the Airflow image, where is it, and how do I update it?

@potiuk CVSS 9,8s are going to be blocked soon for us if they have a fix version, and this does, so I will be unable to use your Airflow image if this happens... And I have been not been able to figure it out, so asking here for some help.

**Comment by potiuk** [comment]
You should ask this question to the security scanner you use. We have no go language in our image and i will not comment nor have any idea where rmyour scanner got the idea Abu tir.  It is likely false positive. The scanners for vulnerabilities often show false positives - that's why we don't t trust them. There is nothing i can do, you can open issue to you e security scanner provider ( i am sure you are easing paid version and gave so e support agreement with them or simply ignore it if you think it is not valid. Can't help more.

**Reply by ahipp13** [reply]
So I looked within the source code for the Airflow image, and it looks like it is downloading a library called "gosu". When I run the "gosu" command, this is what is given back:

gosu version: 1.12 (go1.15.9 on linux/amd64; gc)

It lists the go version that is coming up in the scans. So are you sure this gosu library that is installed in the base Airflow image is not using go? I just want to make sure so when I go to the security scanner team I can provide them with this information.

**Reply by potiuk** [reply]
Feel free to remove it to check - that might be a good solution for you

**Reply by potiuk** [reply]
As I wrote above - we arec ompletely not responsible for the scanner results of yours. Also the images are just "convenience" binaries that make it easy to start but they are not "official" releases of Airflow nor the ASF. The only official releases are source packages that are released officially at https://downloads.apache.org. 

While we put security FRONT and FIRST - all our images are built (at the moment they are built) using latest and most accurate versions of the dependencies at the moment they are pushed - but it's up to the users of them to build and use and extend the images if they feel the need (and their scanners are telling them that) to upgrade to newer versions of images that we pushed at the moment they were released. This is all open-source, you can build the images yourself if you need to use latest version of underlying packages, python versions released or even replace some packages with ones that you vetted by your security team. The instruction on how to build and extend the image are here, https://airflow.apache.org/docs/docker-stack/build.html  tjhe dockerfile to build is here  https://github.com/apache/airflow/blob/main/Dockerfile and there is no more information we have  than you by looking on what is installed, and only you can tell what your scanner is doing. So you are entirely free to get newer versions of packages or system dependencies and build the image as you see fit.

**Reply by potiuk** [reply]
> It lists the go version that is coming up in the scans. So are you sure this gosu library that is installed in the base Airflow image is not using go? I just want to make sure so when I go to the security scanner team I can provide them with this information.

This is really the question to gosu maintainers

**Reply by ahipp13** [reply]
Wanted to reply back to this to give an update. After doing more research within our scanning tool and talking to some people, it turns out that this is not a false positive and that there are 2 versions of go that get installed from the Airflow image you guys provide. The first is go:1.15.9, and this comes onto the pods via the library gosu.  The second, go:1.16.8, comes through the docker version you install, as when you run a "docker version" command you will see this version of go come up:

docker version
Client:
Version:           20.10.9
API version:       1.41
Go version:        go1.16.8
Git commit:        c2ea9bc
Built:             Mon Oct  4 16:03:22 2021
OS/Arch:           linux/amd64
Context:           default
Experimental:      true

The latest version of the docker engine, 23.0, upgrades the go version to fix the CVSS Vulnerabilities. These 2 go versions are related to CVE-2021-38297, which is a CVSS 9.8. Since CVSS 9.8s are getting blocked for my team soon, it looks like we will have to create a custom image in order to continue to use Airflow. Just wanted to share my findings in case somebody else is in a similar situation.

**Reply by potiuk** [reply, marked as answer]
Just be aware that it IS false positive. 

If you actually READ the CVE you linked, the error clearly indicates that it is only for WASM architecture and in the case of docker in image - neither docker go used to build it nor gosu will ever use WASM and JS GOOS (as I understand GOOS to compile any docker binary is linux - but you should double check it and only you can make any decisions there) . 

This is why we never trust such reports and you should neither. You should always check if your case is actually problematic. those security scanners will report all kinds of things that you are not using or which are impossible to trigger in your usage so you should always check yourself if your case is not affected. 

If your security team does not give you an option to check and verify your own way of checking it and ticking the "not applicable" (after you verified it) - it means you are doing work that unnecesarily completess red-tape imposed on you on security team that your company should pay for.

And also this is why if you are concerned of results of security scanner, you should check it yourself.

**Reply by ahipp13** [reply]
Thank you for your response. I am sorry I did not read into it more in depth. I just glanced it over and really did not understand what it meant. I should have did more research.

Currently they do not have a way for us to mark false positives, which is why I am worried. I 100 percent agree with you and am hoping to meet with our security team about this. Worst case scenario I will have to take your image and modify it, but at least there is a solution either way.

I have marked your response as the answer as I feel it is a great way to sum up what people should do if they have a similar situation to me. Thank you again for the help!
